topics/statistics/tutorials/galaxy-ludwig/tutorial.md

```diff
@@ -0,0 +1,468 @@
+
+## MNIST Tabular Structure 
+
+The MNIST dataset consists of images and their corresponding labels. For the purpose of this tutorial, mnist_dataset.csv file is created and contains three columns: image_path, label and, split.
+
+Briefly, the image_path column provides the file paths to the images that will be fed into the deep learning algorithm. The label column contains the correct classifications, ranging from 0 to 9, for the handwritten digits in the images. The split column indicates whether the data should be used for training (0) or testing (2) the model.
+
+![Dataset.csv file format snapshot](../../images/galaxy-ludwig/explain_dataset_format.png "Dataset.csv file format snapshot. features in order: file_path, label, split."){: width="50%" height="10%"}
+ 
+## Configuring Convolutional Neural Networks (CNNs) 
+
+The config.yaml file is crucial as it defines the entire structure of your machine learning experiment. This configuration file tells Ludwig how to process your data, what model to use, how to train it, and what outputs to generate.
+
+The rationale behind the construction of this configuration file for the dataset is as follows:
+
+i) The model takes images as input and uses a stacked convolutional neural network (CNN) to extract features.
+ii) It consists of two convolutional layers followed by a fully connected layer, with dropout applied to the second convolutional layer. The max pooling settings are defined by pool_size = 2 and pool_stride = 2, which are important for reducing the spatial dimensions of the input volume.
+iii) The fully connected layer helps to reduce overfitting.
+iv) The split configuration is used to divide the dataset based on the 'split' column rather than using a random split (which is the default). In this configuration, rows labeled with '0' are included in the training set, while rows labeled with '2' are included in the testing set. If a validation category is needed, rows are labeled with '1'.
+v) The model is trained to classify images into categories (e.g., recognizing digits) and will train for 5 epochs.
+
+![Config.yaml file snapshot](../../images/galaxy-ludwig/explain_config.png "Config.yaml file snapshot."){: width="50%" height="5%"}
+
+
+# Galaxy-Ludwig Tool
+
+Ludwig simplifies the complexities of machine learning by automating essential steps such as data preprocessing, model architecture selection, hyperparameter tuning, and device management. To properly adapt the configuration for different datasets, users should consider the following parameters:
+
+- Input Features: Ensure the type reflects the nature of the data.
+- Encoder: Depending on the input type, this may need to be changed. For example, for image data, consider varying num_filters and filter_size. For text data, you might switch to type: rnn or type: transformer.
```

Comment:
```suggestion
- Encoder: Depending on the input type, this may need to be changed. For example, for image data, consider varying `num_filters` and `filter_size` hyperparameters. For text data, you might switch to type: rnn or type: transformer.
```